docs/dom-testing-library/api-async.md

```diff
@@ -187,3 +187,69 @@ The default `timeout` is `4500ms` which will keep you under
 will detect additions and removals of child elements (including text nodes) in
 the `container` and any of its descendants. It will also detect attribute
 changes.
+
+## `waitForElementToBeRemoved`
+
+```typescript
+function waitForElementToBeRemoved<T>(
+  callback: () => T,
+  options?: {
+    container?: HTMLElement
+    timeout?: number
+    mutationObserverOptions?: MutationObserverInit
+  }
+): Promise<T>
+```
+
+To wait for the removal of element(s) from the DOM you can use
+`waitForElementToBeRemoved`. The `waitForElementToBeRemoved` function is a small
+wrapper around the
+[`MutationObserver`](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver).
+
+Here is an example where the promise resolves with `true` because the element is
+removed:
+
+```javascript
+const el = document.querySelector('div.getOuttaHere')
+waitForElementToBeRemoved(() => document.querySelector('div.getOuttaHere'))
+  .then(() => console.log('Element no longer in DOM'))
+  .catch(err => console.log(`Error you need to deal with: ${err}`))
+
+el.setAttribute('data-neat', true)
+// other mutations are ignored...
+
+el.parentElement.removeChild(el)
+// logs 'Element no longer in DOM'
+```
+
+The callback must return the pre-existing element or array of elements that are expected to be removed.
```

Comment:
I think the explanation that the element(s) must be in the DOM when you call the function should be at the top

Reply:
No problem, I'll make that change now!